### ISSUE TITLE
Poor editing, version chaos

### ISSUE THREAD

**Issue by EpsilonSigma** [csharpfritz/AspTravlerz#4]
On page 58, I read: 
> Finally, you have learned how to customize the commands that interact with your project by using a Program class.

No, I haven't learned anything of the sort. Perhaps the content was removed.

On page 63, I read:
> From  the project.json configuration, recall that...

What project.json? We haven't looked at a project.json file, nor can I find any such file in the sample project we're creating.

On page 66, I read:
> You can start out using the StaticFiles dependencies from Hour 4.

We have not done this in Hour 4. Hence, Listing 5.5 (`app.UseDefaultFiles();` and `app.UseStaticPages();`) doesn't work and I have no idea how to fix the error.

I'm sure there will be a lot more instances like the three above. I think this book has been edited very, very poorly. I'm quite disappointed. I've been waiting for almost two years to learn ASP.NET Core, but all these continuous changes by the ASP.NET team make the task almost impossible. The two books I've purchased were already obsolete by the time they got published, which only added to my confusion.

**Comment by sadqiang** [comment]
Probably because the book had been trimmed from 600 pages to 384 pages.

**Comment by EpsilonSigma** [comment]
How do you know? In any case, the result is that the reader can't follow the instructions in the book. Therefore, the reader can't learn. Therefore, the book is useless and should have never been put on sale. That's how I see it.

**Comment by sadqiang** [comment]
When I read the overview of the book in 2016, it had about 600 pages. But right now it has 384 pages.

**Comment by torr6420** [comment]
Hello,

I agree with you, the book was rushed and not reviewed as it should. The
.NET Core is becoming a trial and error and too complicated and bugged down
with all of these different technologies that I am not going to use
(Lenux/Mac OS), I still like Web Forms better since is more defined and to
the point

I have a problem, I have VS 2017 Community, latest version, I think is
15.5.3 and does not have the project templates for ASP.NET Core Web
Applications (.NET Core) that is mentioned in the book, I have installed
the Professional VS 2017 and is the same.

Have any ideas why?

Thanks

Ray

**Comment by EpsilonSigma** [comment]
The template names in VS 2017 have been changed slightly compared to what you can see in the book. The 'new' Microsoft keeps changing things all the time all over the place—in the spirit of "continuous delivery" perhaps, but the result is endless confusion. All this 'agile' and 'MS loves Linux' madness puts me off.

**Comment by torr6420** [comment]
That is Microsoft for you, a game changer, in constant re-flux, which one
do I use? Is the book useless?

Thanks

**Comment by EpsilonSigma** [comment]
You need to use ASP.NET Core Web Application and on the next page select Core 1.1 in the top right drop-down list. In the top left list you can select either of the two class libraries: either full .NET or .NET Core.

**Comment by csharpfritz** [comment]
Yes, we went through several re-writes before the book was published.

The initial revision focused on project.json, and we thought it was completely removed from the text.  

I am working with the publisher to issue updates.

**Comment by csharpfritz** [comment]
@torr6420 - yes, the project templates have been reorganized and changed by the team.  The book delivers on ASP.NET Core 1.0 and 1.1 features, all of which are also available in ASP.NET Core 2.0

**Comment by torr6420** [comment]
Thank you very much.

**Comment by EpsilonSigma** [comment]
The updates will be of what kind? I have bought both the paper and the electronic book. I guess you won't send me a new copy of the paper version, right?
Also, why learn ASP.NET Core 1.1 now, since it will  be supported only till June 2019, if I recall correctly?

**Comment by csharpfritz** [comment]
All of the features that you learn are still valuable and available in AspNetCore 2.0

I don’t know the process for issuing updates, that’s up to the publisher. 



Jeff

**Comment by torr6420** [comment]
Hello Jeffrey,

I have tried very hard to follow the book, but since nothing in the book
matches with VS 2017 version 15.5.3, Either I am mentally retarded or
something else is wrong with me, I just can't follow the book at all,
forgive me, but that is the truth. This Core stuff is much complicated, and
since there are many different ways to do the same thing, is not clear to
me. I thought this book, like all other Sams Teach Yourself books, took you
by the hand and help you followed along to a complete a project.

I really am thankful that you wrote this book, but it failed completely,
Microsoft constantly keeps changing how projects are created and moving
things around, it is hard to keep up. It is a shame they don't how there
heads are spinning. It is too bad, cause I have always like Microsoft
products.

Ray Torres

**Comment by csharpfritz** [comment]
Where are you disconnected?  What can I help clear up?  I want to help you get on track. 



Jeff

**Comment by torr6420** [comment]
Hello and thanks for offering your help.

I want to start from hour 5, but in the download code there are only 3
files, asptravlerz.csproj, Program.cs and Startup.cs.
It would have been ideal if the solution folder  was going to be included
for each hour?

1- Is the attached screen shot of my New Projects dialog the correct one?
2- Is the New Asp.Net core web app dialog that follows is the correct one?
3- See the Solution Explorer, the wwwroot does not contain any folders or
files?

If not, can you tell me the steps to start from hour 5?

Thank you very much.

Ray

**Comment by torr6420** [comment]
Hello Jeff,

I would like to volunteer to help you diagnose the errors encountered when
trying to follow the book, at no cost to you. Just tell me what you need.

I have a refreshed Win 10 Pro installation and did what's on this page
https://www.microsoft.com/net/learn/get-started/windows?utm_expid=.-Fmi9Q05Ry2oXQgdtPElHw.0&utm_referrer=https%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Faspnet%2Fcore%2Fgetting-started
to install dotnet sdk 2.1.4 and VS 2017 version 15.5.3.

Also if possible, could you do a video of these issues, like the difference
in the new project dialog, the removal IApplicationEnvironment that you
showed in Hour 6, which is where I got stuck cause I don't know what else
to do.

Thanks,

Ray
